lib/util.js

```diff
@@ -915,7 +909,7 @@ function formatMap(ctx, value, recurseTimes, keys) {
 
 function formatWeakSet(ctx, value, recurseTimes, keys) {
   const maxArrayLength = Math.max(ctx.maxArrayLength, 0);
-  const entries = previewWeakSet(value, maxArrayLength + 1);
+  const entries = previewEntries(value).slice(0, maxArrayLength + 1);
```

Comment:
This is pretty bad for performance. The same for the other parts.

Reply:
Yeah, especially the lack of a limit it a bit “:confused:”, but on the other hand I don’t think any of these would typically be hot paths. (In particular, I don’t think we need to make `showHidden: true` fast.)